reliquery/storage.py

```diff
@@ -150,22 +139,11 @@ def dict_from_path(path: StoragePath, dirname: str):
                     with open(self._join_path(entry_path), "r") as f:
                         data[root_key][dirname].append(json.loads(f.read()))
 
-        for d in dirs:
+        for d in data[root_key]:
             dict_from_path(path, d)
```

Comment:
I think we can still modify dict_from_path here to avoid having "side effects." 

https://en.wikipedia.org/wiki/Side_effect_(computer_science)

In general its considered a bit cleaner to not have a function modify state that's not defined outside of it (in this case ```data```). There are a few reasons, but generally these kinds of functions can become confusing and can lead to bugs. I think it may be better to figure out how dict_from_path can just return something that you can add to ```data``` itself from the loop.

Reply:
this applies to the dropbox storage as well, although now i see that you did not originate this code.